### PR TITLE
Reduce the size of the deployed docker image from >350MB to 277MB

### PR DESCRIPTION
**Pull request by segiddins**
comments inline. did this because i was getting frustrated at how much data it took to work while tethering remotely and testing out docker file changes

### CODE REVIEW

**Review by indirect — APPROVED** [review]
looks good!

File: Dockerfile [hunk 1]
```diff
@@ -36,62 +41,73 @@ RUN \
...
+  RAILS_GROUPS=assets SECRET_KEY_BASE=1234 bin/rails assets:precompile
+  rm -r /app/tmp/cache/assets/
+BASH
+
+RUN <<BASH
+  set -ex
+  bundle config set --local without 'development test assets'
   bundle clean --force
+  rm -r /app/tmp/cache/bootsnap/
+  # Precompile bootsnap code for faster boot times, but do it after weve
```
Comment (simi) [inline Dockerfile:102]: `weve` -> `we've` :question:

**Comment by simi** [comment]
@segiddins is this ready? I think rebase will fix the codecov.

**Comment by segiddins** [comment]
@simi rebased, it's ready

**Pull request merged**